### PR TITLE
VZ-7595: Add unit test cases for helm component

### PR DESCRIPTION
**Pull request by karan56625**
Add unit test cases for helm component to increase test coverage.
Package coverage increased from 47.9% to 91.9%

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/helm/helm_component_test.go [hunk 1]
```diff
@@ -814,6 +884,96 @@ func fakeUpgrade(_ vzlog.VerrazzanoLogger, releaseName string, namespace string,
 	return []byte("success"), []byte(""), nil
 }
 
+// TestGetName tests all the getter function of HelmComponent e.g. Namespace, ShouldInstallBeforeUpgrade etc.
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/component/helm/helm_component_test.go:887]: Fix comment to match with func name

File: platform-operator/controllers/verrazzano/component/helm/helm_component_test.go [hunk 2]
```diff
@@ -832,3 +992,703 @@ func fakePreUpgrade(log vzlog.VerrazzanoLogger, client clipkg.Client, release st
...
+		// WHEN  IsAvailable is called
+		// THEN true is returned if component is available
+		{
+			"TestIsAvailableWithAvailableObject",
+			HelmComponent{
+				AvailabilityObjects: &ready.AvailabilityObjects{DeploymentNames: []types.NamespacedName{}},
+			},
+			spi.NewFakeContext(fake.NewClientBuilder().Build(), &v1alpha1.Verrazzano{}, nil, false),
+			"",
+			true,
```
Comment (adiforluls) [inline platform-operator/controllers/verrazzano/component/helm/helm_component_test.go:1026]: Can a test be added for the negative case when IsAvailable returns false?

File: platform-operator/controllers/verrazzano/component/helm/helm_component_test.go (hunk 1 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/component/helm/helm_component_test.go]: fixed!

File: platform-operator/controllers/verrazzano/component/helm/helm_component_test.go (hunk 2 above)
Comment (karan56625) [reply platform-operator/controllers/verrazzano/component/helm/helm_component_test.go:1026]: done

**Review by adiforluls — APPROVED** [review]

**Pull request merged**